src/ApiService/ApiService/onefuzzlib/ScalesetOperations.cs

```diff
@@ -514,6 +514,11 @@ public Async.Task<Scaleset> SetShutdown(Scaleset scaleset, bool now)
 
         //ground truth of existing nodes
         var azureNodes = await _context.VmssOperations.ListInstanceIds(scaleSet.ScalesetId);
+        if (azureNodes is null) {
+            // didn't find scaleset
+            return (false, scaleSet);
+        }
+
```

Comment:
This return is too early, lines 525-530 will clean up our database of nodes that are now deleted in azure.